addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/discovery/HS110DiscoveryService.java

```diff
@@ -0,0 +1,156 @@
+
+    private final InetAddress broadcast;
+
+    private Future<?> scanFuture;
+    private static final int DISCOVER_TIMEOUT_SECONDS = 5;
+    private static final int RECEIVE_JOB_TIMEOUT = 200;
+    private static final int UDP_PACKET_TIMEOUT = RECEIVE_JOB_TIMEOUT - 50;
+
+    public HS110DiscoveryService() throws UnknownHostException {
+
+        super(HS110BindingConstants.SUPPORTED_THING_TYPES_UIDS, DISCOVER_TIMEOUT_SECONDS, false);
+        byte[] addr = { (byte) 0xff, (byte) 0xff, (byte) 0xff, (byte) 0xff };
+        broadcast = InetAddress.getByAddress(addr);
+        discoverPacket = new DatagramPacket(discoverbuffer, discoverbuffer.length, broadcast, 9999);
+
+    }
+
+    @Override
+    protected void startScan() {
+        logger.debug("Start Scan for {}",
+                this.getSupportedThingTypes().stream().map(uid -> uid.toString()).collect(Collectors.joining(" ")));
+        discoverThings();
+
+    }
+
+    @Override
+    protected void stopScan() {
+        logger.debug("Stop Scan for {}",
+                this.getSupportedThingTypes().stream().map(uid -> uid.toString()).collect(Collectors.joining(" ")));
+
+        if (scanFuture != null) {
+            scanFuture.cancel(true);
+        }
+        super.stopScan();
+    }
+
+    private DatagramSocket datagramSocket;
+
+    private void discoverThings() {
+        if (scanFuture == null) {
+            try {
+                datagramSocket = new DatagramSocket(null);
+                datagramSocket.setBroadcast(true);
+                datagramSocket.send(discoverPacket);
+
+                if (logger.isTraceEnabled()) {
+                    String packetContent = new String(discoverPacket.getData(), StandardCharsets.UTF_8);
+                    logger.debug("Discovery package sent: {}", packetContent);
+                }
+
+                datagramSocket.setSoTimeout(UDP_PACKET_TIMEOUT);
+
+                scanFuture = scheduler.scheduleAtFixedRate(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+
+                            datagramSocket.receive(packet);
+
+                            logger.debug("Discovery returned package with length {}", packet.getLength());
+                            if (packet.getLength() > 0) {
+                                detectThing(packet);
+                            }
+
+                            packet.setLength(buffer.length);
+
+                        } catch (SocketTimeoutException e) {
+                            logger.info("No response received in time");
+
+                        } catch (IOException e) {
+                            logger.error("Error during discovery", e);
+                        }
+                    }
+                }, 0, RECEIVE_JOB_TIMEOUT, TimeUnit.MICROSECONDS);
+
+            } catch (IOException e) {
+                logger.error("Error starting discovery", e);
+            } finally {
+                scanFuture = null;
```

Comment:
There are more strange things going on here, you reset the scanFuture here to null, eliminating chances to cancel the `scanFuture`, so it will just keep running.